src/content/roadmap/statelessness/index.md

```diff
@@ -0,0 +1,100 @@
+
+Similarly to history expiry, under state expiry responsibility for storing old state data is removed from individual users and pushed onto other entities such as centralized providers, altruistic community members or more futuristic decentralized solutions such as the Portal Network.
+
+State expiry is still in the research phase and not yet ready to ship. State expiry may well happen later than stateless clients and history expiry because those upgrades make large state sizes easily managebale for the majority of validators.
+
+[Read more on state expiry](https://hackmd.io/@vbuterin/state_size_management#A-more-moderate-solution-state-expiry)
+[Read even more on state expiry](https://hackmd.io/@vbuterin/state_expiry_paths#Option-2-per-epoch-state-expiry)
+
+## Statelessness {#statelessness}
+
+Statelessness is a bit of a misnomer because it does not mean the concept of "state" is eliminated, but it does involve changes to how Ethereum nodes handle state data. Statelessness itself comes in two flavours: weak statelessness and strong statelessness. Weak statelessness enables most nodes to go stateless by putting responsibility for state storage onto a few. Strong statelessness compeltely removes the need for any node to store the full state data. Both weak and strong statelessness offer the following benefits to normal validators:
+
+- nearly instant syncing
+- ability to validate blocks out-of-order
+- nodes able to run with very low hardware requirements (e.g. on phones)
+- nodes can run on top of cheap hard drives because there is no disk reading/writing required
+- compatible with future upgrades to Ethereum's cryptography
+
+Read more on [why it's so important to go stateless](https://dankradfeist.de/ethereum/2021/02/14/why-stateless.html)
+Read the original [stateless client concept notes](https://ethresear.ch/t/the-stateless-client-concept/172)
+
+### Weak Statelessness {#weak-statelessness}
+
+Weak statelessness does involve changes to the way Ethereum nodes verify state changes, but it does not completely eliminate the need for state storage in all nodes on the network. Instead, weak statelessness puts the responsibility for state storage onto block proposers, while all other nodes on the network verify blocks without storing the full state data.
+
+**In weak statelessness proposing blocks requires access to full state data but verifying blocks requires no state data**
+
+For this to happen, [Verkle trees](../verkle-trees) must already have been implemented in Ethereum clients. Verkle trees are a replacement data structure for storing Ethereum state data that allow small, fixed size "witnesses" to the data to be passed between peers and used to verify blocks instead of verifying blocks against local databases. [Proposer-builder separation](../pbs) is also required because this allows block builders to be specialized nodes with more powerful hardware, and those are the ones that require access to the full state data.
```

Comment:
```suggestion
For this to happen, [Verkle trees](../verkle-trees) must already have been implemented in Ethereum clients. Verkle trees are a replacement data structure for storing Ethereum state data that allow small, fixed size "witnesses" to the data to be passed between peers and used to verify blocks instead of verifying blocks against local databases. [Proposer-builder separation](/roadmap/pbs/) is also required because this allows block builders to be specialized nodes with more powerful hardware, and those are the ones that require access to the full state data.
```